File: tests/fixtures/go/slop_restate.go
package main

type HitCounter struct {
	count int
}

func (h *HitCounter) RecordHit() {
	// increment the count
	h.count += 1
}

func (h *HitCounter) RecordBatch(n int) {
	// increment the count
	h.count += n
}

func main() {
	h := &HitCounter{}
	h.RecordHit()
	h.RecordBatch(3)
}

// expect-line: 8 SLOP042
// expect-line: 13 SLOP042

File: tests/fixtures/typescript/slop_restate.ts
class HitCounter {
  count = 0;

  recordHit(): void {
    // increment the count
    this.count += 1;
  }

  recordBatch(n: number): void {
    this.count += n; // increment the count
  }
}

// expect-line: 5 SLOP042
// expect-line: 10 SLOP042
